pkg/envinfo/storage.go

```diff
@@ -56,16 +56,25 @@ func (ei *EnvInfo) GetStorage(name string) (*localConfigProvider.LocalStorage, e
 
 // CreateStorage sets the storage related information in the local configuration
 func (ei *EnvInfo) CreateStorage(storage localConfigProvider.LocalStorage) error {
-	err := ei.devfileObj.Data.AddVolumeMounts(ei.componentSettings.Name, []devfilev1.VolumeMount{
-		{
-			Name: storage.Name,
-			Path: storage.Path,
-		},
-	})
+	// Get all the containers in the devfile
+	containers, err := ei.GetContainers()
 	if err != nil {
 		return err
 	}
 
+	// Add volumeMount to all containers in the devfile
```

Comment:
Doing this because the current behaviour of odo master branch is to add storage to all container components. This is up for discussion in #4105